cinema/views.py

```diff
@@ -56,3 +92,54 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = MovieSession.objects.all()
+        date_str = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+        filters = {}
+        if date_str:
+            date = dt.strptime(date_str, "%Y-%m-%d")
+            filters["show_time__date"] = date
+        if movie:
+            filters["movie__id"] = int(movie)
+        if self.action == "list":
+            queryset = (
+                queryset
+                .select_related("cinema_hall")
+                .annotate(
+                    tickets_available=F("cinema_hall__rows")
+                    * F("cinema_hall__seats_in_row")
+                    - Count("tickets")
+                )
+            ).order_by("id")
+        return queryset.filter(**filters)
+
+
+class OrderPagination(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = "page_size"
+    max_page_size = 100
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderPagination
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        if self.action == "list":
+            queryset = queryset.prefetch_related(
+                "tickets__movie_session__movie",
+                "tickets__movie_session__cinema_hall",
+            ).filter(user=self.request.user)
```

Comment:
The use of `filter(user=self.request.user)` assumes the presence of a `user` field in the Order model. Please verify that this field exists or adjust the filter to match the actual model structure.